tests/unit_tests/Objects/new_variable/test_parameter.py

```diff
@@ -334,4 +339,549 @@ def test_as_data_dict(self, clear, parameter: Parameter):
+
+        # Then
+        result = parameter * test
+        result_reverse = test * parameter
+
+        # Expect
+        assert result.name == expected.name
+        assert result.value == expected.value
+        assert result.unit == expected.unit
+        assert result.variance == pytest.approx(expected.variance)
+        assert result.min == expected.min
+        assert result.max == expected.max
+
+        assert result_reverse.name == expected_reverse.name
+        assert result_reverse.value == expected_reverse.value
+        assert result_reverse.unit == expected_reverse.unit
+        assert result_reverse.variance == pytest.approx(expected_reverse.variance)
+        assert result_reverse.min == expected_reverse.min
+        assert result_reverse.max == expected_reverse.max
+
+    @pytest.mark.parametrize("test, expected, expected_reverse", [
+            (Parameter("test", 0, "", 0.01, -10, 0),    Parameter("name * test", 0.0, "dimensionless", 0.01, -np.Inf, 0), Parameter("test * name", 0, "dimensionless", 0.01, -np.Inf, 0)),
+            (Parameter("test", 0, "", 0.01, 0, 10),     Parameter("name * test", 0.0, "dimensionless", 0.01, 0, np.Inf),  Parameter("test * name", 0, "dimensionless", 0.01, 0, np.Inf))],
+            ids=["zero_min", "zero_max"])
+    def test_multiplication_with_parameter_nan_cases(self, test, expected, expected_reverse):
+        # When
+        parameter = Parameter(name="name", value=1, variance=0.01, min=1, max=np.Inf)
+
+        # Then
+        result = parameter * test
+        result_reverse = test * parameter
+
+        # Expect
+        assert result.name == expected.name
+        assert result.value == expected.value
+        assert result.unit == expected.unit
+        assert result.variance == expected.variance
+
+        assert result_reverse.name == expected_reverse.name
+        assert result_reverse.value == expected_reverse.value
+        assert result_reverse.unit == expected_reverse.unit
+        assert result_reverse.variance == expected_reverse.variance
+
+        if isinstance(result, Parameter):
+            assert result.min == expected.min
+            assert result.max == expected.max
+
+        if isinstance(result_reverse, Parameter):
+            assert result_reverse.min == expected_reverse.min
+            assert result_reverse.max == expected_reverse.max
+
+    @pytest.mark.parametrize("test, expected, expected_reverse", [
+        (DescriptorNumber(name="test", value=2, variance=0.1, unit="cm"), Parameter("name * test", 2, "dm^2", 0.14, 0, 20), Parameter("test * name", 2, "dm^2", 0.14, 0, 20)),
+        (DescriptorNumber(name="test", value=0, variance=0.1, unit="cm"), DescriptorNumber("name * test", 0, "dm^2", 0.1), DescriptorNumber("test * name", 0, "dm^2", 0.1))],
+        ids=["regular", "zero_value"])
+    def test_multiplication_with_descriptor_number(self, parameter : Parameter, test, expected, expected_reverse):
+        # When 
+        parameter._callback = property()
```

Comment:
What is this doing

Reply:
same goes for the tests below

Reply:
No idea! If anyone knows, I'd like to learn. @damskii9992  ?

Reply:
You mean the parameter._callback = property()? It is to use the parameter fixture where the callback is mocked away. So here I just set it to the normal property() value to make it a normal parameter.